tst/unit/kokkos_abstraction.cpp

```diff
@@ -262,6 +262,175 @@ TEST_CASE("par_for loops", "[wrapper]") {
   }
 }
 
+template <class OuterLoopPattern, class InnerLoopPattern>
+bool test_wrapper_nested_3d(OuterLoopPattern outer_loop_pattern,
+                            InnerLoopPattern inner_loop_pattern,
+                            DevExecSpace exec_space) {
+  // Compute the 2nd order centered derivative in x of i+1^2 * j+1^2 * k+1^2
+
+  const int N = 32;
+  ParArray3D<Real> dev_u("device u", N, N, N);
+  ParArray3D<Real> dev_du("device du", N, N, N - 2);
+  auto host_u = Kokkos::create_mirror(dev_u);
+  auto host_du = Kokkos::create_mirror(dev_du);
+
+  // initialize with i^2 * j^2 * k^2
+  for (int n = 0; n < N; n++)
+    for (int k = 0; k < N; k++)
+      for (int j = 0; j < N; j++)
+        for (int i = 0; i < N; i++)
+          host_u(k, j, i) = pow((i + 1) * (j + 1) * (k + 1), 2.0);
+
+  // Copy host array content to device
+  Kokkos::deep_copy(dev_u, host_u);
+
+  // Compute the scratch memory needs
+  const int scratch_level = 0;
+  size_t scratch_size_in_bytes = parthenon::ScratchPad1D<Real>::shmem_size(N);
+
+  // Compute the 2nd order centered derivative in x
+  parthenon::par_outer_for(
+      outer_loop_pattern, "unit test Nested 3D", exec_space, scratch_size_in_bytes,
+      scratch_level, 0, N - 1, 0, N - 1,
+
+      KOKKOS_LAMBDA(parthenon::member_type team_member, const int k, const int j) {
```

Comment:
`parthenon::member_type` is ambiguous outside of a kokkos context, and also kind of a mouthfull. How about
`parthenon::teammbr_t` or something along those lines? For sure we can abbreviate `type` as `t`.

Reply:
`parthenon::team_mbr_t` might work, not too long, distinguishes the abbreviation

Reply:
Yeah, that sounds good.